superset-frontend/plugins/plugin-chart-handlebars/src/plugin/controls/metrics.tsx

```diff
@@ -35,10 +36,10 @@ const percentMetrics: typeof sharedControls.metrics = {
   visibility: isAggMode,
   resetOnHide: false,
   mapStateToProps: ({ datasource, controls }, controlState) => ({
-    columns: datasource?.columns || [],
-    savedMetrics: datasource?.metrics || [],
+    columns: (datasource as Dataset)?.columns || [],
```

Comment:
from my understanding this will be dataset and query and table correct? will there be. type that encompass all three?

Reply:
nvm i see it int he type. lol